Sources/MapboxMaps/Ornaments/OrnamentsManager.swift

```diff
@@ -39,271 +18,165 @@ public enum OrnamentVisibility: String, Equatable {
-     information. See https://docs.mapbox.com/help/how-mapbox-works/attribution/
-     for details.
+    internal init(view: OrnamentSupportableView, options: OrnamentOptions) {
+        self.options = options
 
-     TODO: Implement logo and attribution ornaments
-     */
+        // Logo View
+        logoView = LogoView(logoSize: .regular)
+        logoView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(logoView)
 
-    internal init(for view: OrnamentSupportableView, withConfig ornamentConfig: OrnamentConfig) {
-        super.init()
-        self.view = view
-        self.ornamentConfig = ornamentConfig
+        // Scalebar View
+        scalebarView = MapboxScaleBarOrnamentView()
+        scalebarView.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(scalebarView)
 
-        ensureTelemetryOptOutExists()
-    }
+        // Compass View
+        compassView = MapboxCompassOrnamentView(visibility: options.compassVisibility)
+        compassView.translatesAutoresizingMaskIntoConstraints = false
+        compassView.tapAction = { [weak view] in
+            view?.compassTapped()
+        }
+        view.addSubview(compassView)
 
-    internal func addOrnament(_ ornamentType: OrnamentType,
-                              at position: OrnamentPosition,
-                              visibility: OrnamentVisibility) {
-        let ornament = Ornament(view: nil, type: ornamentType, position: position, visibility: visibility)
-        ornamentConfig = ornamentConfig.with(ornament: ornament)
-    }
+        // Info Button
+        attributionButton = MapboxInfoButtonOrnament()
+        attributionButton.translatesAutoresizingMaskIntoConstraints = false
+        view.addSubview(attributionButton)
 
-    internal func addOrnament(_ ornamentView: UIView, at position: OrnamentPosition) {
-        let ornament = Ornament(view: nil, type: .customOrnament(value: ornamentView), position: position)
-        ornamentConfig = ornamentConfig.with(ornament: ornament)
-    }
+        super.init()
 
-    internal func removeOrnament(_ ornamentView: UIView) {
-        ornamentConfig = OrnamentConfig(ornaments: ornaments.filter {
-            $0.view != ornamentView
-        })
-    }
+        updateOrnaments()
 
-    internal func removeOrnament(at position: OrnamentPosition) {
-        ornamentConfig = OrnamentConfig(ornaments: ornaments.filter {
-            $0.position != position
-        })
+        // Subscribe to updates for scalebar and compass
+        view.subscribeCameraChangeHandler { [scalebarView, compassView] (cameraOptions) in
+            if let zoom = cameraOptions.zoom,
+               let center = cameraOptions.center {
+                scalebarView.metersPerPoint = Projection.getMetersPerPixelAtLatitude(
+                    forLatitude: center.latitude,
+                    zoom: Double(zoom))
+            }
+            if let bearing = cameraOptions.bearing {
```

Comment:
nit: newline

Reply:
before this line? why? do we have guidelines on vertical whitespace?

Reply:
No guidelines around this. But just looking over this, this felt harder to read as opposed to having a newline. It's a nit so not going to hold the PR